src/lib/incapsula.ts

```diff
@@ -0,0 +1,48 @@
+import * as bluebird from 'bluebird';
+import * as http from 'superagent';
+import { buildServiceResponse, ServiceResponse } from '../lib/service';
+import urlModule = require('url');
+
+const BASE_URL = 'https://my.incapsula.com';
+const BASE_PATH = '/api/prov/v1';
+
+export function flushSite(siteID: string, incapsulaApiID: string, incapsulaApiKey: string): bluebird<ServiceResponse> {
+  return buildServiceResponse(
+    http
+      .post(
+        `${BASE_URL}${BASE_PATH}/sites/cache/purge?api_id=${incapsulaApiID}&api_key=${incapsulaApiKey}&site_id=${siteID}`
+      )
+      .accept('application/json')
+  );
+}
+
+export function flushURLs(
+  siteID: string,
+  url: string,
+  incapsulaApiID: string,
+  incapsulaApiKey: string
+): bluebird<ServiceResponse> {
+  return buildServiceResponse(
+    http
+      .post(
+        `${BASE_URL}${BASE_PATH}/sites/cache/purge?api_id=${incapsulaApiID}&api_key=${incapsulaApiKey}&site_id=${siteID}&resource_pattern=${
+          urlModule.parse(url).pathname
+        }`
+      )
+      .accept('application/json')
+  );
+}
+
+export function flushHostname(
```

Comment:
remove this code as incapsula doesn't support flush of resource by hostname.

Reply:
👍